### PR TITLE
fix(TDP-2573): quick switch between apps

### PR DESCRIPTION
**Pull request by ncomont**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-2573

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
**- [?] Docker configuration files for config-std or config-cloud profiles are impacted**

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

**(Optional) What is the current behavior?**
(Additional information to the Jira)

**(Optional) What is the new behavior?**
(Additional information to the Jira)

**(Optional) Other information**:

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/437/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/437/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/438/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/438/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/447/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/447/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/457/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/457/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/468/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/468/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/472/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/472/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/483/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/483/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/485/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/485/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/487/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/487/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/490/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/490/)

File: dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/AppSettingsService.java [hunk 1]
```diff
@@ -65,6 +65,7 @@ public AppSettings getSettings() {
 
         // populate appSettings actions dictionary (key: actionId, value: action)
         getSettingsStream(actionsProviders, actionsConfigurers) //
+                .filter(actionSettings -> actionSettings.getFiltered() == null || !actionSettings.getFiltered())
```
Comment (fhuaulme) [inline dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/AppSettingsService.java]: I'd prefer a method named "isEnabled" that returns a `boolean` (not a `Boolean` not you don't have the need to check the `null` value). Also renamed `filtered` filed to `enabled` and set default value to true in the builder.

File: dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/actions/api/ActionDropdownSettings.java [hunk 2]
```diff
@@ -87,7 +87,8 @@ public static Builder from(final ActionDropdownSettings actionSettings) {
                 .bsStyle(actionSettings.getBsStyle()) //
                 .items(actionSettings.getItems()) //
                 .dynamicAction(actionSettings.getDynamicAction()) //
-                .staticActions(actionSettings.getStaticActions());
+                .staticActions(actionSettings.getStaticActions()) //
+                .filtered(actionSettings.getFiltered());
```
Comment (fhuaulme) [inline dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/actions/api/ActionDropdownSettings.java]: filtered -> enabled.

File: dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/actions/api/ActionDropdownSettings.java [hunk 3]
```diff
@@ -135,6 +138,11 @@ public Builder items(final String items) {
             return this;
         }
 
+        public Builder filtered(final Boolean filtered) {
```
Comment (fhuaulme) [inline dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/actions/api/ActionDropdownSettings.java]: Use a `boolean` here.

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/492/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/492/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/493/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/493/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/496/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/496/)

**Pull request merged**